src/libs/actions/Transaction.ts

```diff
@@ -820,17 +819,75 @@ function changeTransactionsReport(transactionIDs: string[], reportID: string) {
             key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${transactionThreadReportID}`,
             value: {[movedAction?.reportActionID]: null},
         });
-
+        const selfDMReportID = findSelfDMReportID();
+        const currentTime = DateUtils.getDBTime();
+        const selfDMReport = buildOptimisticSelfDMReport(currentTime);
+        const selfDMCreatedReportAction = buildOptimisticCreatedReportAction(currentUserEmail ?? '', currentTime);
```

Comment:
We should only do this if `selfDMReportID` doesn't exist yet. Otherwise, we should use the existing selfDM report:
```suggestion
        const existingSelfDMReportID = findSelfDMReportID();
        if (!existingSelfDMReportID) {
            const currentTime = DateUtils.getDBTime();
            const selfDMReport = buildOptimisticSelfDMReport(currentTime);
            const selfDMCreatedReportAction = buildOptimisticCreatedReportAction(currentUserEmail ?? '', currentTime);
        }
```